src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs

```diff
@@ -478,6 +478,13 @@ SyntaxKind.ImplicitObjectCreationExpression or
                 return CreateAdjustSpacesOperation(0, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
             }
 
+            // paramName!!
+            if (currentToken.Kind() == SyntaxKind.ExclamationExclamationToken &&
+                currentToken.Parent.IsKind(SyntaxKind.Parameter))
+            {
+                return CreateAdjustSpacesOperation(0, AdjustSpacesOption.ForceSpacesIfOnSingleLine);
+            }
```

Comment:
Consider also adding a test to `SyntacticClassifierTests.cs ` and `SemanticQuickInfoSourceTests.cs`. This way, the PR covers all IDE basics (formatting, classifier, quick info, since completion is N/A).